backend/ops_api/ops/resources/services_component.py

```diff
@@ -0,0 +1,195 @@
+from datetime import date
+from functools import partial
+
+import marshmallow_dataclass as mmdc
+from flask import Response, current_app, request
+from flask_jwt_extended import get_jwt_identity, verify_jwt_in_request
+from models import ContractAgreement, OpsEventType, ServicesComponent
+from models.base import BaseModel
+from ops_api.ops.base_views import BaseItemAPI, BaseListAPI, handle_api_error
+from ops_api.ops.resources.services_component_schemas import (
+    PATCHRequestBody,
+    POSTRequestBody,
+    QueryParameters,
+    ServicesComponentItemResponse,
+)
+from ops_api.ops.utils.api_helpers import get_change_data, update_and_commit_model_instance
+from ops_api.ops.utils.auth import ExtraCheckError, Permission, PermissionType, is_authorized
+from ops_api.ops.utils.events import OpsEventHandler
+from ops_api.ops.utils.response import make_response_with_headers
+from ops_api.ops.utils.user import get_user_from_token
+from sqlalchemy import select
+from typing_extensions import override
+
+ENDPOINT_STRING = "/services-components"
+
+
+def sc_associated_with_contract_agreement(self, id: int, permission_type: PermissionType) -> bool:
+    jwt_identity = get_jwt_identity()
+    try:
+        contract_agreement_id = request.json["contract_agreement_id"]
+        agreement_stmt = select(ContractAgreement).where(ContractAgreement.id == contract_agreement_id)
+        contract_agreement = current_app.db_session.scalar(agreement_stmt)
+
+    except KeyError:
+        sc_stmt = select(ServicesComponent).where(ServicesComponent.id == id)
```

Comment:
these 2 lines could be combined like: `sc = current_app.db_session.get(ServiceComponent, id)`

Reply:
done